src/megatron/energon/wrappers/batch_dataset.py

```diff
@@ -104,9 +75,13 @@ def __init__(
         self.batcher_config = batcher_config
         self.drop_last = drop_last
         self.error_handler = error_handler
-        self._sample_index = SampleIndex(worker_config, src=self)
-        self._generator_sample_keys = [None] * max(self.worker_config.num_workers, 1)
-        self._generator_offsets = [None] * max(self.worker_config.num_workers, 1)
+
+        self.reset_state_own()
```

Comment:
In general, I wonder if `reset_state_own` should only be called in the worker? I.e. not in the init at all. Because the state actually only exists in the worker, right?